lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -236,6 +266,27 @@ struct ConversionOpConversion : public OpConversionPattern<ConversionOp> {
 // Statement Conversion
 //===----------------------------------------------------------------------===//
 
+struct AssignOpConversion : public OpConversionPattern<BlockingAssignOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(BlockingAssignOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto resultType = typeConverter->convertType(op.getDst().getType());
+    StringAttr nameAttr;
+    if (auto varOp =
+            llvm::dyn_cast_or_null<VariableOp>(op.getDst().getDefiningOp()))
+      nameAttr = varOp.getNameAttr();
+    // rewriter.setInsertionPointAfterValue(op.getDst());
+    auto from = rewriter.getRemappedValue(op.getDst());
+    rewriter.replaceOpWithNewOp<hw::WireOp>(from.getDefiningOp(), resultType,
+                                            adaptor.getSrc(), nameAttr,
+                                            hw::InnerSymAttr{});
```

Comment:
We have `moore.struct_create` to declaration a struct, and the union same as. So we need to `if` branches to estimate.